packages/nitro-rpc-client/src/serde.ts

```diff
@@ -14,12 +14,20 @@ const jsonRpcSchema = {
   properties: {
     jsonrpc: { type: "string" },
     id: { type: "uint32" },
+  },
+  optionalProperties: {
     result: {
-      properties: {},
+      nullable: true,
+    },
+    error: {
+      properties: {
+        code: { type: "int32" },
+        message: { type: "string" },
+      },
       additionalProperties: true,
+      nullable: true,
```

Comment:
According to the [AJV documentation](https://ajv.js.org/json-type-definition.html#union-keyword), it seems this is not possible using JTD (which is what we're using instead of JSON schema). We could add logic external to the JTD if we want to enforce only one of "result" or "error" is present.